### PR TITLE
run: parallel tests capability

### PR DESCRIPTION
**Pull request by TharushiJay**
Running tests in parallel test runs to help speed up the checks in terms of time.

Link to redmine ticket: https://redmine.openinfosecfoundation.org/issues/3357

Changes from last PR: 
- Fixed the test count issue
- Modified the display output

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -619,30 +624,30 @@ def run(self):
             check_value = self.check()
             if check_value["check_sh"]:
                 return check_value
-
+
```
Comment (jasonish) [inline run.py:627]: nit: white space added

**Review by jasonish — CHANGES_REQUESTED** [review]

**Comment by jasonish** [comment]
Looking really good.

Its not immediately apparent to me why, but the command line option, `--debug-failed` is no longer working. To see what should happen, tweak a test so that it fails. For example, I made this change:
```
diff --git a/tests/dns-eve-type-filtering/test.yaml b/tests/dns-eve-type-filtering/test.yaml
index 24dc330..dd95651 100644
--- a/tests/dns-eve-type-filtering/test.yaml
+++ b/tests/dns-eve-type-filtering/test.yaml
@@ -17,7 +17,7 @@ checks:
         event_type: "dns"
   - filter:
       filename: only-a.json
-      count: 4
+      count: 5
       match:
         event_type: "dns"
         dns.rrtype: "A"
```

If you run suricata-verify against Suricata using the SV master branch, you'll get some output at the end like:
```
Failed tests debug output:
- Test dns-eve-type-filtering:
  - /home/jason/oisf/code/suricata-verify/tests/dns-eve-type-filtering/output/stdout
    - [Not dumping file that won't utf-8 decode]
  - /home/jason/oisf/code/suricata-verify/tests/dns-eve-type-filtering/output/stderr
    - [Not dumping file that won't utf-8 decode]
...
```
But with this branch, this output doesn't appear to get trigger at all.

**Comment by jasonish** [comment]
An enhancement to this feature would be to add a `-j` command line parameter to provide the number of jobs to run. The default behaviour should be what you have here already, but `-j` would allow me tweak it.  Maybe I only want to have 2 (`-j 2`) parallel tests going at a time.. Or even one (`-j 1`).

**Comment by TharushiJay** [comment]
> Looking really good.
> 
> Its not immediately apparent to me why, but the command line option, `--debug-failed` is no longer working. To see what should happen, tweak a test so that it fails. For example, I made this change:
> 
> ```
> diff --git a/tests/dns-eve-type-filtering/test.yaml b/tests/dns-eve-type-filtering/test.yaml
> index 24dc330..dd95651 100644
> --- a/tests/dns-eve-type-filtering/test.yaml
> +++ b/tests/dns-eve-type-filtering/test.yaml
> @@ -17,7 +17,7 @@ checks:
>          event_type: "dns"
>    - filter:
>        filename: only-a.json
> -      count: 4
> +      count: 5
>        match:
>          event_type: "dns"
>          dns.rrtype: "A"
> ```
> 
> If you run suricata-verify against Suricata using the SV master branch, you'll get some output at the end like:
> 
> ```
> Failed tests debug output:
> - Test dns-eve-type-filtering:
>   - /home/jason/oisf/code/suricata-verify/tests/dns-eve-type-filtering/output/stdout
>     - [Not dumping file that won't utf-8 decode]
>   - /home/jason/oisf/code/suricata-verify/tests/dns-eve-type-filtering/output/stderr
>     - [Not dumping file that won't utf-8 decode]
> ...
> ```
> 
> But with this branch, this output doesn't appear to get trigger at all.

Thanks for the feedback @jasonish. I'll work on the changes! 👍

**Comment by TharushiJay** [comment]
Changes continued in #389.

**Pull request closed without merging**